File: 2022/js/day06.js
const fs = require("fs");

const sol = (len) => {
  const seq = fs.readFileSync("input.txt").toString().trim();
  for (let i = 0; i < seq.length; i++) {
    if (new Set(seq.substring(i, i + len)).size === len) return i + len;
  }
};

console.log(sol(4), sol(14));

File: 2022/python/day06.py
def sol(length):
    res = 0
    with open("input.txt") as file:
        seq = file.read().strip()
        for i in range(len(seq)):
            if len(set(seq[i:i+length])) == length:
                res = i + length
                break
    return res

if __name__ == "__main__":
    print(sol(4), sol(14))
